packages/vitest/src/node/config.ts

```diff
@@ -273,7 +273,7 @@ export function resolveConfig(
         ?? resolve(resolved.root, file),
     ),
   )
-  resolved.coverage.exclude.push(...resolved.setupFiles.map(file => relative(resolved.root, file)))
+  resolved.coverage.exclude.push(...resolved.setupFiles.map(file => `${resolved.coverage.allowExternal ? '**/' : ''}${relative(resolved.root, file)}`))
```

Comment:
this was added because the test for `src/another-setup.ts` was failing. Because test-exclude will try to resolve absolute path if allowExternal is set to true I added `**/` prefix. 
I don't think this is a good solution as it may exclude some files we didn't wish to exclude (e.g. if the setup file is called `src/index.ts`)

Reply:
This makes sense but is indeed not ideal. Let's leave this open for now and proceed with other changes.

Reply:
I think this change is fine. Users should defined `setupFiles` with absolute path if this causes issues.

Reply:
Wouldn't actually something like this make more sense?
```suggestion
  resolved.coverage.exclude.push(...resolved.setupFiles.map(file => resolved.coverage.allowExternal ? file : relative(resolved.root, file)))
```